phrase_extraction_evaluation.py

```diff
@@ -0,0 +1,92 @@
+# Class to evaluate model output
+
+import numpy as np
+from typing import List
+from sentence_transformers import SentenceTransformer
+
+embedding_model = SentenceTransformer("all-MiniLM-L6-v2")
+
+class STSEvaluation:
+    treshhold=0.75
+
+    @staticmethod
+    def get_embedding(phrase: str) -> np.ndarray:
+        return embedding_model.encode([phrase])[0]
+
+    @staticmethod
+    def get_embeddings(phrases: List[str]) -> List[np.ndarray]:
+        return embedding_model.encode(phrases)
+    
+    @staticmethod
+    def cosine_similarity(a: np.ndarray, b: np.ndarray) -> float:
+        dot_product = np.dot(a, b)
+        a_magnitude = np.linalg.norm(a)
+        b_magnitude = np.linalg.norm(b)
+        if a_magnitude == 0 or b_magnitude == 0:
+            return 0.0
+        return dot_product / (a_magnitude * b_magnitude)
+    
+    @staticmethod
+    def make_similarity_matrix(predictions, labels):
+        similarity_matrix=np.zeros([len(predictions),len(labels)])
+        for i in range(0, len(predictions)):
+            for j in range(0,len(labels)):
+                similarity_matrix[i][j]=STSEvaluation.cosine_similarity(predictions[i],labels[j])
+        return similarity_matrix
+    @staticmethod
+    def average_cosine_score(similarity_matrix) -> float:
+        result = 0
+        for j in range(0, len(similarity_matrix[0])):
+            best_score=0
+            for i in range(0,len(similarity_matrix)):
+                if best_score<similarity_matrix[i][j]:
+                    best_score=similarity_matrix[i][j]
+            result+=best_score
+        return result / len(similarity_matrix[0])
+    
+    @staticmethod
```

Comment:
Evo i ovde jos jednog primera kako moze da bude bolji kod:
def matched_labels(similarity_matrix) -> float:
    return sum(any(similarity_matrix[i][j] >= STSEvaluation.treshhold for i in range(len(similarity_matrix))) for j in range(len(similarity_matrix[0]))) / len(similarity_matrix[0])